test/test_mkldnn.py

```diff
@@ -260,20 +260,40 @@ def test_adaptive_avg_pool2d(self):
         self.assertEqual(x1.grad, x2.grad.to_dense())
 
     def test_batch_norm2d(self):
-        N = torch.randint(3, 10, (1,)).item()
-        C = torch.randint(3, 100, (1,)).item()
-        x = torch.randn(N, C, 35, 45, dtype=torch.float32) * 10
-
-        # TODO: support training
-        for train in [False]:
-            bn = torch.nn.BatchNorm2d(C).float().train(train)
-            mkldnn_bn = mkldnn_utils.to_mkldnn(copy.deepcopy(bn))
-            self.assertEqual(
-                bn(x),
-                mkldnn_bn(x.to_mkldnn()).to_dense())
+        x = torch.randn(64, 3, 35, 45, dtype=torch.float32) * 10
+
+        for train in [True, False]:
+            # TODO: support none affine
+            for affine in [True]:
+                for track_running_stats in [True, False]:
+                    bn = torch.nn.BatchNorm2d(
+                        3,
+                        affine=affine,
+                        track_running_stats=track_running_stats).float().train(train)
+                    if (train or not track_running_stats):
+                        mkldnn_bn = copy.deepcopy(bn)
```

Comment:
Could you explain why in this case you don't send the module to mkldnn?

Reply:
For training case, module's parameters awalys a dense tensor, not need to call **mkldnn_utils.to_mkldnn**

Reply:
Ho ok. But why is `track_running_stats` here?